tests/run-pass/fs.rs

```diff
@@ -39,6 +39,8 @@ fn main() {
     // Reading until EOF should get the whole text.
     file.read_to_end(&mut contents).unwrap();
     assert_eq!(bytes, contents.as_slice());
+    // Cloning a file should be successful
+    file.try_clone().unwrap();
```

Comment:
Is there anything useful we can test on the returned FD to make sure that this actually worked?

Reply:
I think  we might need a function that compares two files in terms of content and metadata. In this case, I suppose the metadata should be identical except for the dates that should be larger for the cloned file. And of course the cloned file descriptor should be strictly larger than the original one.